### PR TITLE
Improved error messages and cleaned up code

### PR DESCRIPTION
**Pull request by philippjfr**
Some attempt at:

1. Improving error messages so they state what the wrong value actually was (this is not always obvious)
2. Clean up line lengths
3. Clean up of spacing between arguments.

### CODE REVIEW

File: param/__init__.py [hunk 1]
```diff
@@ -956,25 +964,30 @@ class Boolean(Parameter):
...
             if val is not True and val is not False and val is not None:
-                raise ValueError("Boolean '%s' must be True, False, or None."%self.name)
+                raise ValueError("Boolean '%s' must be True, False, or None, "
+                                 "not %s."% (self.name, val))
         else:
             if not isinstance(val,bool):
-                raise ValueError("Boolean '%s' only takes a Boolean value."%self.name)
+                raise ValueError("Boolean '%s' only takes a Boolean value, "
+                                 "not type %s."
+                                 % (self.name, type(val)))
```
Comment (jbednar) [inline param/__init__.py:986]: Is the type more informative here than the value? Should it maybe show both (here and below)? Not sure.

File: param/__init__.py [hunk 2]
```diff
@@ -1379,7 +1399,8 @@ def _validate(self, val):
             return
 
         if not isinstance(val, list):
-            raise ValueError("List '%s' must be a list."%(self.name))
+            raise ValueError("List '%s' must be a list, not %s."
```
Comment (jbednar) [inline param/__init__.py:1402]:
```suggestion
            raise ValueError("List '%s' must be a list, not an object of type %s."
```

**Review by jbednar — APPROVED** [review]

File: param/__init__.py (hunk 1 above)
Comment (philippjfr) [reply param/__init__.py:986]: Yeah, I'm vacillating here. I think it should just show the value.

File: param/__init__.py [hunk 3]
```diff
@@ -956,25 +964,30 @@ class Boolean(Parameter):
...
         if self.allow_None:
-            if not isinstance(val,bool) and val is not None:
-                raise ValueError("Boolean '%s' only takes a Boolean value or None."
-                                 %self.name)
+            if not isinstance(val, bool) and val is not None:
+                raise ValueError("Boolean '%s' only takes a Boolean value "
+                                 "or None, not type %s."
+                                 % (self.name, type(val)))
 
             if val is not True and val is not False and val is not None:
```
Comment (philippjfr) [inline param/__init__.py:979]: @jbednar This seems like dead code, am I wrong? The `not isinstance(val, bool) and val is not None` should catch this already, right?

Comment (jbednar) [reply param/__init__.py:979]: I agree. There does not seem to be any possible value of `val` for which this second clause would fire that would not already have fired for the first one.

**Pull request merged**